### PR TITLE
refactor: move component styling to styled

### PR DESCRIPTION
**Pull request by haxxmaxx**
This gets the styling out of the component files. Styled should be faster as well. There is some logic in there but this should be covered by rendering tests. If we like, the styling can be on top in each compnent file, but outside the component fn.

### CODE REVIEW

File: src/table/styles.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  overflow: 'hidden',
+  position: 'absolute',
+  whiteSpace: 'nowrap',
+  width: '1px',
+});
+
+// ---------- FooterWrapper ----------
+
+export const StyledFooterWrapper = styled(Paper, {
+  shouldForwardProp: (prop) => prop !== 'tableTheme',
```
Comment (LiKang6688) [inline src/table/styles.js:18]: should be `shouldForwardProp: (prop) => prop !== 'tableTheme' && prop !== 'theme', `?

File: src/table/styles.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    borderLeft: !paginationNeeded && '1px solid rgb(217, 217, 217)',
+  },
+  '& td, th': {
+    fontSize: bodyCellStyle.fontSize,
+    padding: bodyCellStyle.padding,
+  },
+}));
+
+export const StyledBodyRow = styled(TableRow, {
+  shouldForwardProp: (prop) => prop !== 'bodyCellStyle',
```
Comment (LiKang6688) [inline src/table/styles.js:69]: should be `shouldForwardProp: (prop) => prop !== 'hover && prop !== 'bodyCellStyle'',`?

File: src/table/styles.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
+import { styled, Paper, TableContainer, TableRow, TableSortLabel, TableBody, Select, IconButton } from '@mui/material';
+
+// ---------- AnnounceWrapper ----------
```
[the remark this answers is not in the document]
Comment (haxxmaxx) [reply src/table/styles.js:3]: alphabetically

File: src/table/styles.js (hunk 2 above)
Comment (haxxmaxx) [reply src/table/styles.js:69]: no hover is a property actually used by the component, so it should be passed

File: src/table/styles.js (hunk 1 above)
Comment (haxxmaxx) [reply src/table/styles.js:18]: no theme is the mui theme, no need to suppress it here

Comment (LiKang6688) [reply src/table/styles.js:18]: right, I saw it is used by the component itself

**Pull request merged**